### DISCUSSION TITLE
How to retrieve data from an array containing elements of different types

### DISCUSSION THREAD

**Discussion by Jengro777** [vlang/v#21278] [General]
`["Hello", "World", 0, 1, null]`

**Comment by JalonSolov** [comment]
Since it's not valid V, I'm not sure what to say.

In V, all elements of an array must be the same type.

I suppose you could have an array of sum types, but that's the only way you could construct that array in V in the first place.

**Reply by Jengro777** [reply]
I don't want to build this type of array, but rather to retrieve data from this type of array.
Due to the upstream return of such data, I need to use V language to process this type of array

**Reply by JalonSolov** [reply]
You'll probably have to write your own parser, unless you can get `x.json2` to recognize it as JSON data, then you can use https://modules.vlang.io/x.json2.html#raw_decode to parse it for you.

**Comment by Jengro777** [comment]
`[[["Hello", "Hi", 0, 1, null],["World", "Peter", 0, 1, null]],null,"success"]`


What should I do if I want to obtain "Hollo Word" for this type of data?

**Reply by Jengro777** [reply]
![image](https://github.com/vlang/v/assets/35259040/2a17f8e2-8ec6-4dfb-ab45-cfbee5d1003e)

**Comment by Jengro777** [comment]
```
import x.json2

fn main() {

    mut text_arr := json2.raw_decode('[[["Hello", "Hi", 0, 1, null],["World", "Peter", 0, 1, null]],null,"success"]')!
    
    text1 := text_arr.arr()
    text2 := text1[0].arr()

    mut arr := []string{}
    for i in text2 {
        text3 := i.arr()
        text4 := text3[0].arr()
        eprintln("text4 is :$text4")
        arr << text4[0].str()    
    }
    eprintln("arr is :$arr")

}

```
